performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/UITests.swift

```diff
@@ -71,33 +71,63 @@ class UITests: XCTestCase {
     #else
       if !image {
         XCUIRemote.shared.press(.down)
+        XCUIRemote.shared.press(.down)
+        XCUIRemote.shared.press(.up)
         sleep(1)
         XCTAssert(app.cells["Classify Image"].hasFocus, "Reached invalid state.")
       }
       XCUIRemote.shared.press(.select)
     #endif
   }
 
-  func download(timeout: TimeInterval = 10) throws {
+  func download(image: Bool = false, timeout: TimeInterval = 10) throws {
     #if os(iOS)
       XCTAssert(app.buttons["Download Image"].isHittable, "Reached invalid state.")
       app.buttons["Download Image"].tap()
     #else
-      XCUIRemote.shared.press(.up)
-      sleep(1)
-      XCTAssert(app.cells["Download Image"].hasFocus, "Reached invalid state.")
+      if !image {
+        XCUIRemote.shared.press(.up)
+        XCUIRemote.shared.press(.up)
+        sleep(1)
+        XCTAssert(app.cells["Download Image"].hasFocus, "Reached invalid state.")
+      }
       XCUIRemote.shared.press(.select)
     #endif
+    if image {
+      XCTAssert(app.images.firstMatch.waitForExistence(timeout: timeout),
+                "Failed to retrieve image.")
+    }
+  }
 
-    XCTAssert(app.images.firstMatch.waitForExistence(timeout: timeout), "Failed to retrieve image.")
+  func upload(image: Bool = false, timeout: TimeInterval = 10) throws {
+    #if os(iOS)
+      XCTAssert(app.buttons["Upload Image"].isHittable, "Reached invalid state.")
+      app.buttons["Upload Image"].tap()
+    #else
+      if !image {
+        XCUIRemote.shared.press(.down)
+        XCUIRemote.shared.press(.down)
+        sleep(1)
+        XCTAssert(app.cells["Upload Image"].hasFocus, "Reached invalid state.")
+      }
+      XCUIRemote.shared.press(.select)
+    #endif
   }
 
   func testAllViews() throws {
```

Comment:
I like this idea of tests. But if it does not hurt, can we split tests by purpose? Eg. separate test for Download, upload and Classify? When something fails, it would be easier to plot which test has failed.

Reply:
Given that I still see value in testing all the functions working together, I will add additional tests that run first to separate out the functionality (e.g `testDownload`), and only then test all functionality together.

Reply:
Done ✅